osdep/compiler.h

```diff
@@ -22,6 +22,10 @@
 #define MP_UNUSED
 #endif
 
+#if defined(__STDC_VERSION__) && (__STDC_VERSION__ < 202311L) && !defined(thread_local)
```

Comment:
could you make it work for c++ too?

Reply:
I'm not too well versed on c++ (and wasn't even aware mpv had any until now), but looking at this https://en.cppreference.com/w/cpp/language/storage_duration; it seems like `thread_local` is a keyword there already since `c++11`. So it probably "just works".

Or is it not working for you for some reason?

Reply:
It's fine. I was actually thinking to not define it on C++, but it would already not do that, because we already know that STDC is not defined there. So my bad, for not thinking this through.